src/openlifu/vf/virtual_fit.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from dataclasses import dataclass, field
+from typing import Optional, Tuple
+
+import numpy as np
+import xarray as xa
+
+from openlifu.geo import Point
+from openlifu.plan import TargetConstraints
+from openlifu.xdc import Transducer
+
+
+@dataclass
+class VirtualFit:
+    """
+    VirtualFit class.
+
+    Represents the virtual fitting algorithm which consists in
+    finding the optimal transducer transform (position and orientation)
+    given an input MRI volume and the associated target.
+    """
+    pitch_range: Tuple[int, int] = (10, 40)
+    """The pitch range for the grid search."""
+
+    pitch_step: int = 3
+    """The pitch step for the grid search."""
+
+    yaw_range: Tuple[int, int] = (-5, 25)
+    """The yaw range for the grid search."""
+
+    yaw_step: int = 3
+    """The yaw step for the grid search."""
+
+    search_range_units: str = "deg"
+    """Search grid units."""
+
+    steering_limits: Tuple[TargetConstraints] = field(default_factory=list)
+    """Defines the accepteable range for a target in the transducer space, usually (lat, ele, ax)."""
+
+    blocked_elems_threshold: float = 0.1
+    """How much blocked elements are acceptable."""
+
+    volume: xa.Dataset = field(default_factory=xa.Dataset)
+    """The MRI volume on which to optimize the position."""
```

Comment:
Just confirming, this volume is assumed to be already in transducer coordinates, correct?

Reply:
Actually no... I do still have this question... is the volume meant to be in lat, ele, ax coords here?

Reply:
Sorry I keep confusing myself. I think almost certainly the answer is yes: the volume is in lat, ele, ax coordinates, and so is the target.

Reply:
It is indeed in (lat, ele, ax), if we don't satisfy this we would need to resample to that coordinate before virtual fitting (as for `Protocol.calc_solution`). Just updated the docstrings in a new commit. https://github.com/OpenwaterHealth/OpenLIFU-python/pull/162/commits/529404002778fe2ccaa028c3b67a51f3d17cee03